fabfile.py

```diff
@@ -10,65 +10,120 @@
+    put, get, local, sudo, run, require)
+from fabric.contrib.files import (exists, is_link)
+from fabric.api import (
+    abort, cd, env, execute, hide, prefix, settings, task as fab_task)
 from fabric.colors import yellow, cyan, red, green
-from fabric.contrib.files import *
-
-
-""" The various services used by the stack """
-SERVICES = [
-    'postgres',
-    'smtp',
-    'imap',
-    'piwik',
-    'sentry',
-    'memcached',
-    'redis',
-    'mysql',
-]
+
+
 ELASTICSEARCH_VERSION = '5.2.2'
 
 
-def load_service_configs():
-    """ allows to break a service network configuration into many files.
-    Suppose we have a platform instance's configuration specified in
-    platform_xxx.rc. In that file, we can refer to other configuration files
-    with the pattern [service]_config = [service_name]
-    for each service defined in SERVICES above.
-    This function will then look for the file configs/[service]/[service_name].rc
-    and import all the "key = value" pairs therein in the current environment,
-    with keys prefixed with [service]_.
-    Note that variables already defined in the environment, or in the platform's
-    .rc file, will override those found in from service config files."""
-    from fabric.state import env
+def combine_rc(rc_filename, overlay=None):
+    """Take a rc filename, load it as fabric would.
+
+    If it specifies an _extends value, consider this file
+    to be an overlay of the named file."""
     from fabric.main import load_settings
-    for service in SERVICES:
-        config_name = env.get(service + "_config", "default")
-        if config_name is None:
-            continue
-        fname = join('configs', service, config_name + ".rc")
-        if os.path.exists(fname):
-            service_config = load_settings(fname)
-            for k, v in service_config.iteritems():
-                full_key_name = "_".join((service, k))
-                if full_key_name not in env:
-                    env[full_key_name] = v
+    assert os.path.exists(rc_filename), "Can't find " + rc_filename
+    service_config = load_settings(rc_filename)
+    if '_extends' in service_config:
+        fname = service_config['_extends']
+        # We want fname to be usable both on host and target.
+        # Use project-path-relative names to that effect.
+        if fname.startswith('~/'):
+            path = dirname(__file__)
+            if env.host_string != 'localhost':
+                path = env.get('projectpath', path)
+            fname = join(path, fname[2:])
+        else:
+            fname = join(dirname(rc_filename), fname)
+        service_config = combine_rc(fname, service_config)
+    if overlay is not None:
+        service_config.update(overlay)
+    service_config.pop('_extends', None)
+    service_config.pop('', None)
+    return service_config
 
 
-load_service_configs()
+def filter_global_names(rc_data):
+    """Some keys in rc files are prefixed with * or _ for ini conversion purposes"""
```

Comment:
Add: Returns the list of names in rc file that are not global nor private. Just keys. Unless the value of the key is `__delete_key__``